### PR TITLE
Use v1 API to create time series with legacyName

### PR DESCRIPTION
**Pull request by hakontro**

### CODE REVIEW

File: src/main/scala/com/cognite/spark/datasource/TimeSeriesRelation.scala [hunk 1]
```diff
@@ -259,15 +288,15 @@ class TimeSeriesRelation(config: RelationConfig)(val sqlContext: SQLContext)
...
+  def baseTimeSeriesUrl(project: String, version: String = "v1"): Uri =
     uri"${baseUrl(project, version, config.baseUrl)}/timeseries"
 
   override def schema: StructType = structType[TimeSeriesItem]
 
   override def toRow(t: TimeSeriesItem): Row = asRow(t)
 
   override def listUrl(): Uri =
-    uri"${config.baseUrl}/api/0.5/projects/${config.project}/timeseries"
+    uri"${config.baseUrl}/api/v1/projects/${config.project}/timeseries"
```
Comment (wjoel) [inline src/main/scala/com/cognite/spark/datasource/TimeSeriesRelation.scala]: Looks like we could use `baseTimeSeriesUrl(config.project, "v1")` here as well?

File: src/test/scala/com/cognite/spark/datasource/TimeSeriesRelationTest.scala [hunk 2]
```diff
@@ -64,6 +64,7 @@ class TimeSeriesRelationTest extends FlatSpec with Matchers with SparkTest {
           .sql(s"""
                 |select '$updatedDescription' as description,
                 |name,
+                |name as legacyName,
```
Comment (wjoel) [inline src/test/scala/com/cognite/spark/datasource/TimeSeriesRelationTest.scala]: Remove this from tests, we don't want to change the external schema and it doesn't look like we are. :+1:

File: src/test/scala/com/cognite/spark/datasource/TimeSeriesRelationTest.scala [hunk 3]
```diff
@@ -115,6 +116,7 @@ class TimeSeriesRelationTest extends FlatSpec with Matchers with SparkTest {
          |select '$insertDescription' as description,
          |isString,
          |concat('TEST_', name) as name,
+         |concat('TEST_', name) as legacyName,
```
Comment (wjoel) [inline src/test/scala/com/cognite/spark/datasource/TimeSeriesRelationTest.scala]: Remove this from tests, we don't want to change the external schema and it doesn't look like we are. :+1:

File: src/test/scala/com/cognite/spark/datasource/TimeSeriesRelationTest.scala [hunk 4]
```diff
@@ -145,6 +147,7 @@ class TimeSeriesRelationTest extends FlatSpec with Matchers with SparkTest {
            |select description,
            |isString,
            |name,
+           |name as legacyName,
```
Comment (wjoel) [inline src/test/scala/com/cognite/spark/datasource/TimeSeriesRelationTest.scala]: Remove this from tests, we don't want to change the external schema and it doesn't look like we are. :+1:

File: src/test/scala/com/cognite/spark/datasource/TimeSeriesRelationTest.scala [hunk 5]
```diff
@@ -33,6 +33,7 @@ class TimeSeriesRelationTest extends FlatSpec with Matchers with SparkTest {
       .sql(s"""
          |select '$initialDescription' as description,
          |concat('TEST', name) as name,
+         |concat('TEST', name) as legacyName,
```
Comment (wjoel) [inline src/test/scala/com/cognite/spark/datasource/TimeSeriesRelationTest.scala]: Remove this from tests, we don't want to change the external schema and it doesn't look like we are. :+1:

File: src/main/scala/com/cognite/spark/datasource/TimeSeriesRelation.scala (hunk 1 above)
Comment (hakontro) [reply src/main/scala/com/cognite/spark/datasource/TimeSeriesRelation.scala]: We can! Nice.

File: src/test/scala/com/cognite/spark/datasource/TimeSeriesRelationTest.scala (hunk 2 above)
Comment (hakontro) [reply src/test/scala/com/cognite/spark/datasource/TimeSeriesRelationTest.scala]: Done! Had make a new case class to get correct schema, but no big deal I guess.

**Pull request merged**